File: src/main/java/org/cloudfoundry/credhub/domain/CertificateCredentialFactory.java
package org.cloudfoundry.credhub.domain;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import org.cloudfoundry.credhub.credential.CertificateCredentialValue;
import org.cloudfoundry.credhub.entity.Credential;

@Component
public class CertificateCredentialFactory {

  private final Encryptor encryptor;

  @Autowired
  CertificateCredentialFactory(final Encryptor encryptor) {
    super();
    this.encryptor = encryptor;
  }

  public CertificateCredentialVersion makeNewCredentialVersion(
    final Credential certificateCredential,
    final CertificateCredentialValue credentialValue
  ) {
    final CertificateCredentialVersion version = new CertificateCredentialVersion(credentialValue, encryptor);
    version.setCredential(certificateCredential);

    return version;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/domain/CertificateCredentialFactory.kt
package org.cloudfoundry.credhub.domain

import org.cloudfoundry.credhub.credential.CertificateCredentialValue
import org.cloudfoundry.credhub.entities.Credential
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class CertificateCredentialFactory @Autowired
internal constructor(private val encryptor: Encryptor) {

    fun makeNewCredentialVersion(
        certificateCredential: Credential,
        credentialValue: CertificateCredentialValue
    ): CertificateCredentialVersion {
        val version = CertificateCredentialVersion(credentialValue, certificateCredential.name!!, encryptor)
        version.credential = certificateCredential

        return version
    }
}
